### PR TITLE
fuzzing support

### PR DESCRIPTION
**Pull request by 0x34d**
Adding fuzzing to finding memory-corruption-related bugs.

Hello coturn team,
Can you check this pr harness suite for creating harnesses and compiling harnesses?
Any other thoughts on adding a new interface for fuzzing support ?

Signed-off-by: 0x34d <ajsinghyadav00@gmail.com>

### CODE REVIEW

**Comment by ggarber** [comment]
@0x34d Can you clarify why the Google LLC copyright and where the corpus is taken from?

Also, what was the result of running the fuzzing?

**Comment by 0x34d** [comment]
> @0x34d Can you clarify why the Google LLC copyright and where the corpus is taken from?

1. It was a copy of oss-fuzz PR.
2. FuzzStun_seed_corpus are from ```rfc5769check.c```  ``` reqstc,reqltc,respv4,respv6,```
3. FuzzStunClient_seed_corpus are from [Wireshark capture](https://www.cloudshark.org/captures/4686b8430477)

**Comment by 0x34d** [comment]
> Also, what was the result of running the fuzzing?

Currently, ClusterFuzz on the job.
Well, we all will know.

File: fuzzing/Fuzzing.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+- Compile Everything.
+```
+cd coturn/fuzzing/
+make
+```
+
+- Run the Fuzzer.
+```
+cd input/
```
Comment (eakraly) [inline fuzzing/Fuzzing.md:51]: Can this be done in makefile? As one of the existing targets or a new one?

File: fuzzing/Fuzzing.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+How-to Guide for fuzzing in localhost.
+
+
+- Export flags.
+
+####
+LibFuzzer with AddressSanitizer
+
+```
+export CC=clang
```
Comment (eakraly) [inline fuzzing/Fuzzing.md:11]: Thinking out loud: having these 3 options in a script (or maybe as separate makefile targets) to make it easier to run the fuzzer/automate it (when run not a part of a fuzzing cluster which I suppose injects those settings)?

**Comment by 0x34d** [comment]
@eakraly Can you review it now? 
I'm leaving ```CC``` ```CXX``` for a variety of fuzzers.

- libfuzzer
- AFL++
- honggfuzz
- centipede

File: fuzzing/build.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+   DIR=build/fuzzing
+   if [ $1 == '0' ]
+   then
+      ./$DIR/FuzzStun   $DIR/FuzzStun_Corpus/   $DIR/FuzzStun_seed_corpus
+   else
+      ./$DIR/FuzzStunClient   $DIR/FuzzStunClient_Corpus/   $DIR/FuzzStunClient_seed_corpus
+   fi   
+}
+
+if [ $1 == "ASan" ]
```
Comment (eakraly) [inline fuzzing/build.sh:30]: Running build.sh with no args results in 
```
$ ./build.sh 
./build.sh: line 30: $1: unbound variable
```

**Review by ggarber — APPROVED** [review]

**Comment by ggarber** [comment]
Thank you very much for the work here @0x34d  (and @eakraly ).  I think it is fine to merge it.

I will try to add executing the fuzzer to the CI scripts when I get some time.

**Pull request merged**